Data-Structures/src/main/java/tree/BinaryTree.java

```diff
@@ -62,22 +62,31 @@ public BinaryTree(Node<T> root){
     }
   }
 
-  public ArrayList<T> breadthFirst(){
-    ArrayList<T> ans = new ArrayList<>();
-    Queue<T> q = new Queue();
-    Node<T> current = this.root;
-    while(current != null){
-      q.enqueue(current.value);
+  // TODO:
+  public ArrayList<Node> breadthFirst(){
+    ArrayList<Node> ans = new ArrayList<>();
+    Queue<Node> q = new Queue();
+    q.enqueue(this.root);
+    while(!ans.isEmpty()){
+      Node current = q.dequeue();
+      ans.add(current);
       if(current.left != null){
-        q.enqueue(current.left.value);
+        q.enqueue(current.left);
       }
       if(current.right != null){
-        q.enqueue(current.right.value);
+        q.enqueue(current.right);
       }
-      ans.add(q.dequeue());
     }
     return ans;
   }
 
+  public int findMaxValue(Node root){
+    int max = -1;
```

Comment:
Food for thought: will the user know `-1` means the tree is empty or think that that is the max value in the tree?